geops/trafimage-maps
# Add apiKeyName api parameter

See ROKAS-68

# How to

The application should behave the same. All api requests should use the api key 'key'. However, it's possible to globally change the api key name.


Test WKP with modifications
- Open the Network tab in the developer tools
- Run https://github.com/geops/trafimage-maps/pull/813 locally
=> The basemap does not load
- Search for 'api_key' in the Network tab of the developer tools
=> `https://maps.geops.io/styles/base_bright_v2/style.json...` should be listed with an apiKey `12345`
- Activate layer "Pünktlichkeit"
- Search for anything
=> The parameter `key` (not `api_key`) should be used with a apiKey `12345` for
`..trajectory_collection?...` 
`https://api.geops.io/stops/v1/?&q=Ber&key=12345`
=> Additionally `Unhandled Rejection (TypeError): Cannot read property 'slice' of undefined` appears as no items exist

Test WKP without modifications
- Use the test app from https://github.com/geops/trafimage-maps/pull/814
=> All layers/services should work

Test Casa Map with modifications
- Open the Casa Map example in the test app from https://github.com/geops/trafimage-maps/pull/822
- Open the Network tab in the developer tools and search for 'api_key'
=> The urls should be 
`https://api.geops.io/routing/v1/TEST/`
`https://maps.geops.io/styles/base_bright_v2/style.json`

Test Casa Map without modifications
- Use the test app for Casa Map from https://github.com/geops/trafimage-maps/pull/814
=> All layers should work

# Others



- [x] It's not a hack or at least an unauthorized hack :).
- [x] Everything in ticket description has been fixed. (Together with https://github.com/geops/trafimage-maps/pull/797)
- [x] The author of the MR has made its own review before assigning the reviewer.
- [ ] IE11 tested.
- [x] The title means something for a human being.
- [x] Labels applied. if it's a release? a hotfix?
